src/backend/gporca/libgpos/src/error/CFSimulator.cpp

```diff
@@ -158,7 +158,10 @@ CFSimulator::Init()
 void
 CFSimulator::Shutdown()
 {
-	CMemoryPool *mp = m_mp;
+	if (NULL == m_fsim)
+		return;
+
+	CMemoryPool *mp = m_fsim->m_mp;
```

Comment:
> Why not to do like (and shutdown of other classes, related to this patch, like at CMessageRepository)?
> Why does we need to store the mp at separate variable and previously cleanup and null the m_fsim, and after that
> delete mp?

Here it is done to reduce the diff.

Reply:
> Why does we need to store the mp at separate variable and previously cleanup and null the m_fsim, and after that
> delete mp?

GPOS_DELETE uses a memory pool.

Reply:
> seems the next block would be more readable

Just wrapping an empty line will increase the diff again.

Reply:
if this code is valid (from the first message):
```
CMemoryPoolManager::GetMemoryPoolMgr()->Destroy(m_flsim->mp);
GPOS_DELETE(m_fsim);
m_fsim = NULL
```

Seems it's worth to refactor (at this and other classes). Yes it will increase the diff, but what is the purpose of reducing the diff? It may be more complex to push a refactor changes to upstream, and in addition this the single approach, used at Shutdown methods is easier to review